tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -519,8 +519,17 @@ private void InitWafAndInstrumentations(bool fromRemoteConfig = false)
                 _waf = _wafInitResult.Waf;
                 oldWaf?.Dispose();
                 Log.Debug("Disposed old waf and affected new waf");
+
+                // if we not in an RC scenario - we may have some rule data hard coded into the WAF rules file
+                // - the system test use this technique
+                // in this case, on start up, we don't want to overwrite the rule data
+                // but if we've already received data from remote config then we do want to overwrite on startup
                 var ruleData = _remoteConfigurationStatus.RulesDataByFile.SelectMany(x => x.Value).ToList();
-                UpdateWafWithRulesData(ruleData);
+                if (fromRemoteConfig || ruleData.Count > 0)
```

Comment:
I maybe misunderstand the point of this test, but shouldn't it be `!fromRemoteConfig`? i.e. you want to use the hardcoded data in that scenario?